.binder/postBuild

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# This script is called in a binder context. When this script is called, we are
+# inside a git checkout of the scikit-learn/scikit-learn repo. This script is
+# going to generate notebooks from the python examples.
+ 
+# Back up content we need from the scikit-learn repo
+TMP_CONTENT_DIR=/tmp/scikit-learn
+mkdir -p $TMP_CONTENT_DIR
+cp -r examples .binder $TMP_CONTENT_DIR
+rm -rf * .*
```

Comment:
That one is scary even if it's in a `postBuild` script. I can always imagine something running this file by accident. Is there a way to be more specific about the wildcard?

Reply:
@rth I pushed a commit to avoid running outside of `repo2docker` context. That should avoid people running `postBuild` locally and deleting all their files in their current directory.

@betatim: to figure out whether I am in a `repo2docker` context I am using the following lines below, let me know if there is a more robust way to do that:
```
if [[ -z "$NB_USER" ]] || [[ "$USER" != "$NB_USER" ]]; then
    echo "This script is supposed to run in a repo2docker context."
    echo "Exiting because this script can remove files if run locally.\n"
    exit 1
fi
```

Reply:
Good question. I don't think we currently have a good way for scripts during build or run time to detect that they are inside a repo2docker run.

I'd probably check the `REPO_DIR` environment variable to know "I am in repo2docker land" as well as `/.dockerenv` which is a docker specific thing to tell if you are inside a container or not.

Reply:
Thanks a lot for your input @betatim. I think it's fine to only check whether we are inside a docker container actually so I went for checking `/.dockerenv`.

More details:
* I googled a bit and it looks like `/.dockerenv` may not be reliable in the long-term. For example [this](https://github.com/moby/moby/issues/18355#issuecomment-220484748) seems to advise against relying on `/.dockerenv`. Having said that, `./dockerenv` is still used (at the time of writing) in `docker/libnetwork` see [this](https://github.com/docker/libnetwork/blob/92d1fbe1eb0883cf11d283cea8e658275146411d/drivers/bridge/setup_bridgenetfiltering.go#L160) and this comment was from 3 years ago, so maybe that's robust enough for our purposes.
* Another way is to check if we are running inside a docker container is to check if docker is in `/proc/1/cgroup` e.g. a small adaptation of [this](https://stackoverflow.com/a/41559867):
```bash
INSIDE_DOCKER=$(grep -q docker /proc/1/cgroup && echo "yes" || echo "no")
```
* Overall `/.dockerenv` feels simpler and robust enough for our purposes.